### PR TITLE
Fix Ubuntu (latest) tests

### PR DESCRIPTION
**Pull request by klytje**
As mentioned in #2812, the CI system was previously failing for ubuntu (latest). Though the problem seems to have solved itself, this PR ensures it will not happen again.

### CODE REVIEW

File: .github/workflows/matrix.py [hunk 1]
```diff
@@ -28,7 +28,7 @@
 os_release_list = [
     'ubuntu-20.04',
     'windows-latest',
-    'macos-latest',
+    'macos-latest',
```
Comment (llimeht) [inline .github/workflows/matrix.py:31]: Would be nice to not load in a whitespace diff

**Review by llimeht — APPROVED** [review]
Thanks, looks good (other than the whitespace change)

File: .github/workflows/matrix.py (hunk 1 above)
Comment (klytje) [reply .github/workflows/matrix.py]: fixed

**Comment by klytje** [comment]
Merging is blocked because `main` is protected; is this due to a lack of permissions from my side or because I should target a different base to merge into?

**Comment by butlerpd** [comment]
Thanks @klytje! Nice work. The locked main branch is a hopefully brief holding pattern as we sort out what should be merged to `main` and what should be merged into `release_6.0.0` first and then pulled into `main` when `release_6.0.0` is merged back to `main`.  

I'm realizing as I write this the current  PR could be at risk since the `ci.yml` files have diverged and currently we have to resolve conflicts whenever merging `release_6.0.0` into `main`. I'm feeling this could lead to an accidental overwrite of this fix?

Ideally we should have a common `ci.yml`  which I think @krzywon has suggested?  Thoughts?

**Pull request merged**